### ISSUE TITLE
A means to register all classes under a directory

### ISSUE THREAD

**Issue by benlesh** [ember-cli/ember-cli#2145]
I talked with @stefanpenner about this, and I think I was too busy eating at the time to digest his answer. So if this feature already exists, I apologize.

If I add a new directory under `app/`, it would be great if there were a means to automatically register things under that directory with the container.  For example:

``` sh
└ app
 ├ adapters
 ├ controllers
 ├ queries
 │ ├ foo.js
 │ └ bar.js
 └ templates
```

where `foo` and `bar` export default basic Ember classes.  And I'm able to get to them with `this.container.lookupFactory('query:foo');` or `this.container.lookupFactory('query:bar');`.

Ideally, this would be something that's configurable via Ember-CLI, so I could say, "register all classes under [directory], then inject those classes into their similarly named routes, or inject them into all controllers" and so on.

Basically, it would be something that would automatically produce an initializer to reduce busy work.
